website/docs/r/policy_ipsec_vpn_session.markdown

```diff
@@ -80,9 +80,12 @@ The following arguments are supported:
 * `peer_id` - (Optional) Peer ID to uniquely identify the peer site. The peer ID is the public IP address of the remote device terminating the VPN tunnel. When NAT is configured for the peer, enter the private IP address of the peer.
 * `local_endpoint_path` - (Required) Policy path referencing Local endpoint. In VMC, Local Endpoints are pre-configured the user can refer to their path using `data nsxt_policy_ipsec_vpn_local_endpoint` and using the "Private IP1" or "Public IP1" values to refer to the private and public endpoints respectively.
 * `rule` - (Optional) Bypass rules for this IPSec VPN Session. `sources` and `destinations` are list of `ipv4-cidr-block`, `Action` value must be set to `BYPASS` or `PROTECT`. Only applied for Policy Based VPN Session. 
+  * `nsx_id` - (Optional) The NSX ID of the bypass rule for this IPSec VPN Session. If set, this ID will be used to create the resource.
```

Comment:
Allowing to specify `nsx_id` in sub-resource (rule in this case) has potential for issues, since there is no `ForceNew` functionality specific to the sub-resource. It would be painful to implement partial update for rules if we choose to do so - we'll need to implement `ForceNew` functionality ourselves on rule level. To avoid these kind of challenges, I suggest to make `nsx_id` read-only.

Reply:
Thanks Anna, i've changed the nsx_id in IPSecVpnRule schema to use getFlexNsxIDSchema and set readOnly to true

Reply:
Thanks, consequently this documentation line needs to be removed or moved to `attributes` under `rule`